src/ranged.cpp

```diff
@@ -2771,19 +2774,18 @@ void target_ui::action_switch_mode()
     }
 
     menu.query();
+    bool refresh = false;
     if( menu.ret >= firing_modes_range.first && menu.ret < firing_modes_range.second ) {
         // gun mode select
         const std::map<gun_mode_id, gun_mode> all_gun_modes = relevant->gun_all_modes();
         int skip = menu.ret - firing_modes_range.first;
         for( std::pair<gun_mode_id, gun_mode> it : all_gun_modes ) {
-            if( it.second.flags.count( "REACH_ATTACK" ) ) {
-                continue;
-            }
             if( skip-- == 0 ) {
                 relevant->gun_set_mode( it.first );
                 break;
             }
         }
+        refresh = true;
```

Comment:
You may want to restrict this to abort only when switching to melee mode. There's no need to abort aiming when switching between e.g. single shot and burst fire, or burst fire and an underbarrel attachment.